src/api/lib/statistics_calculations.rb

```diff
@@ -19,9 +21,12 @@ def self.packages(limit, timelimit, prj_filter, pkg_filter)
   private_class_method :packages
 
   def self.projects(limit, timelimit, prj_filter)
-    Project.where(updated_at: timelimit..Time.now).
-      where('name REGEXP ?', prj_filter).
-      order('updated_at DESC').
+    projects = Project.all
+    projects = projects.where(updated_at: timelimit..Time.now) if timelimit
+    if prj_filter
+      projects = projects.where('name REGEXP ?', prj_filter)
+    end
```

Comment:
Nitpick: I would make this a guard clause, given that you already use one one line above.

Reply:
done